src/main/java/org/prebid/server/auction/AmpRequestFactory.java

```diff
@@ -221,37 +227,64 @@ private BidRequest overrideParameters(BidRequest bidRequest, HttpServerRequest r
             }
         }
 
-        final Site updatedSite = overrideSite(bidRequest.getSite(), request);
-        final Imp updatedImp = overrideImp(bidRequest.getImp().get(0), request);
+        final String requestTargeting = request.getParam(TARGETING_REQUEST_PARAM);
+        final Targeting targeting = parseTargeting(requestTargeting);
+        final ObjectNode targetingNode = (ObjectNode) readTargeting(requestTargeting);
+
+        final Site updatedSite = overrideSite(bidRequest.getSite(), request, targeting);
+        final Imp updatedImp = overrideImp(bidRequest.getImp().get(0), request, targetingNode);
         final Long updatedTimeout = overrideTimeout(bidRequest.getTmax(), request);
-        final User updatedUser = overrideUser(bidRequest.getUser(), gdprConsent);
+        final User updatedUser = overrideUser(bidRequest.getUser(), gdprConsent, targeting);
         final Regs updatedRegs = overrideRegs(bidRequest.getRegs(), ccpaConsent);
+        final ExtRequest updatedExtBidRequest = overrideExtBidRequest(bidRequest.getExt(), targeting);
 
         final BidRequest result;
         if (updatedSite != null || updatedImp != null || updatedTimeout != null || updatedUser != null
-                || updatedRegs != null) {
+                || updatedRegs != null || updatedExtBidRequest != null) {
             result = bidRequest.toBuilder()
                     .site(updatedSite != null ? updatedSite : bidRequest.getSite())
                     .imp(updatedImp != null ? Collections.singletonList(updatedImp) : bidRequest.getImp())
                     .tmax(updatedTimeout != null ? updatedTimeout : bidRequest.getTmax())
                     .user(updatedUser != null ? updatedUser : bidRequest.getUser())
                     .regs(updatedRegs != null ? updatedRegs : bidRequest.getRegs())
+                    .ext(updatedExtBidRequest != null ? updatedExtBidRequest : bidRequest.getExt())
                     .build();
         } else {
             result = bidRequest;
         }
         return result;
     }
 
-    private Site overrideSite(Site site, HttpServerRequest request) {
+    private JsonNode readTargeting(String requestTargeting) {
+        try {
+            return requestTargeting != null ? mapper.mapper().readTree(requestTargeting) : null;
```

Comment:
`requestTargeting` looks to be the same value as input parameter in `parseTargeting`. However here it is not decoded prior to parsing. Is it intentional difference?

Reply:
It is mistake, fixed